src/metabase/metabot/util.clj

```diff
@@ -42,34 +42,37 @@
           (str/replace #" " "_")))
 
 (defn- aliases
-  "Create a seq of aliases to be used to make model columns more human friendly"
+  "Create a seq of aliases to be used to make model columns more human friendly.
+  If this is a native query or for any other reason we don't get aliases we just fall back to '*'"
   [{:keys [dataset_query result_metadata]}]
-  (let [alias-map (update-vals
-                   (into {} (map (juxt :id :display_name) result_metadata))
-                   (fn [v] (cond-> v
-                             (str/includes? v " ")
-                             normalize-name)))
-        {:keys [query]} (let [qp (qp.reducible/combine-middleware
-                                  (vec qp/around-middleware)
-                                  (fn [query _rff _context]
-                                    (add/add-alias-info
-                                     (#'qp/preprocess* query))))]
-                          (qp dataset_query nil nil))
-        {:keys [fields]} query]
-    (for [field fields
-          :let [[_ id m] field
-                alias (::add/desired-alias m)]]
-      [alias (alias-map id)])))
+  (if dataset_query
```

Comment:
all models (and all cards) must have a `dataset_query`. In the db it is marked `not null`. This seems to want to branch on the `:type` of the dataset_query and not the presence/absence of it.

```clojure
{:id 1335,
 :name "native",
 :dataset_query
 {:type :native, ;; this is the determiner of native vs mbql
  :native {:query "select id, ean, title, category, vendor, price, rating, created_at from products",
           :template-tags {}},
  :database 1}}
```